src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    if (this[0] !== undefined && this[0] !== null) {
+      string += this[0];
+    }
```

Comment:
This check is redundant here, just move it to the main cycle and check all elements on undefined or null